operator/cmd/mesh/install.go

```diff
@@ -236,9 +240,103 @@ func InstallManifests(iop *v1alpha12.IstioOperator, force bool, dryRun bool, res
 		return iop, err
 	}
 
+	if dryRun {
+		currentObj, err := reconciler.GetObject(iop)
+		if err != nil {
+			return iop, err
+		}
+		obj, err := object.ParseYAMLToK8sObject([]byte(iopStr))
+		if err != nil {
+			return iop, err
+		}
+		expectedObj := obj.UnstructuredObject()
+		result, err := showDifferences(currentObj, expectedObj)
+		if err != nil {
+			return iop, err
+		}
+		// Display the results to the user
+		l.Print(result)
+	}
 	return iop, saveIOPToCluster(reconciler, iopStr)
 }
 
+func showDifferences(existing, target *unstructured.Unstructured) (string, error) {
+	existingYAML := []byte("")
+	if existing != nil {
+		var err error
+		existingYAML, err = object.NewK8sObject(existing, nil, nil).YAML()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	targetYAML := []byte("")
+	if target != nil {
+		var err error
+		targetYAML, err = object.NewK8sObject(target, nil, nil).YAML()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	result := "\n\nConfiguration differences:\n"
+	result += util.YAMLReducedDiff(string(existingYAML), string(targetYAML), 3)
+
+	existingManifest, err := ManifestFromString(string(existingYAML))
+	if err != nil {
+		return "", err
+	}
+	targetManifest, err := ManifestFromString(string(targetYAML))
+	if err != nil {
+		return "", err
+	}
+
+	result += "\nManifest differences...\n"
+	maniDiff, err := compare.ManifestDiff(existingManifest, targetManifest, false)
+	if err != nil {
+		return "", err
+	}
+	result += maniDiff
+	return result, nil
+}
+
+// ManifestString returns manifest based on given a yaml string,
+func ManifestFromString(iopsYAML string) (string, error) {
```

Comment:
function name is misleading, this should really be something like ManifestFromIOP